### ISSUE TITLE
systemd-tmpfiles: Failed to adjust quota for subvolume "/var/lib/portables": Transport endpoint is not connected

### ISSUE THREAD

**Issue by daandemeyer** [systemd/systemd#15896]
**systemd version the issue has been seen with**
> v245

**Used distribution**
 > Arch Linux

**Expected behaviour you didn't see**
 > No errors when building Arch Linux mkosi image

**Unexpected behaviour you saw**
 > systemd-tmpfiles fails with the error in the title

**Steps to reproduce the problem**

`sudo mkosi --default .mkosi/mkosi.arch -f`

So I finally took some time to dig into this error. When building an Arch Linux mkosi image using systemd's Arch mkosi configuration, pacman's systemd-tmpfiles hook consistently failed with "error: command failed to execute correctly". I hacked a bunch of extra logging into pacman and finally got the following error from `systemd-tmpfiles --create`:

```
"Failed to adjust quota for subvolume "/var/lib/portables": Transport endpoint is not connected"
```

Investigating this lead me to `btrfs_subvol_auto_qgroup_fd` which fails with `ENOTCONN`. `ENOTCONN` is not used as a synthetic errno in btrfs-util.c so I assume one of the btrfs ioctls is returning `ENOTCONN`. I'll see if I can hack an `strace` into the pacman hook when it's running inside the mkosi image to see exactly which ioctl call is failed.

(As an aside, error backtraces would be extremely useful to have when debugging stuff like this. Some kind of macro (e.g. `TRACE`) that stores a call stack whenever a syscall fails or a function returns a synthetic errno and is printed along with the error message when `log_error_errno` is called. glibc has a `backtrace` which we could use to get a stacktrace.)

**Comment by poettering** [comment]
hmm, i figure this is caused by not having quota on on the btrfs file system? if you enable quota, does the message go away? if so, we should probably treat ENOTCONN like ENOPROTOOPT

**Comment by poettering** [comment]
So this is caused by this:

https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=8a36e408d40606e21cd4e2dd9601004a67b14868

**Comment by poettering** [comment]
Fix is in #15922. Would be great if you could give it a whirl.

(And no need to enable quota anymore, with that patch)
